### ISSUE TITLE
Pretrained models

### ISSUE THREAD

**Issue by fzliu** [mozilla/DeepSpeech#758]
I read from the wiki that a LibriSpeech + Fisher + TED-LIUM training run is currently in the works.

When that is finished, will the model be made public? I'm eager to try it out in a couple of applications, and I imagine other developers are as well.

**Comment by 96imranahmed** [comment]
Second that, a pretrained model would be very useful - just so we have a baseline with which we can finetune (happy to contribute the model back to the world if I get any improvement)

**Comment by furkanarius** [comment]
I would also love to see a pretrained model we can use.

Furkan AKYUREK
R & D Engineer


[http://sign1.trtworld.com/trtworldlogo.png]

Ahmet Adnan Saygun Caddesi. No:83
Ulus / Be?ikta? / ?stanbul - Turkey

Phone:  +90 212 326 30 00 / Ext : 3127
Mob:    +90 506 396 71 18


[cid:ICON_19X19_WEB_da8453d8-15e5-4269-8a0d-aa34326f5800.jpg]<https://www.trtworld.com> [cid:ICON_19X19_YOUTUBE_c24221d2-0f98-4804-994d-e0392b3a0d21.jpg] <https://www.youtube.com/c/trtworld>  [cid:ICON_19X19_FACEBOOK_78226783-072b-46e4-9739-74f38ce232eb.jpg] <https://www.facebook.com/trtworld>  [cid:ICON_19X19_TWITTER_c19878df-3a28-4faa-a2ab-f47439ce4a14.jpg] <https://twitter.com/trtworld>  [cid:ICON_19X19_INSTAGRAM_7c44092c-e90b-48d9-876e-3edba5146fb3.jpg] <https://instagram.com/trtworld>
<http://www.trtworld.com><http://www.trtworld.com/july15/><https://www.trtworld.com>[cid:IN_AUGUST_TRTWORLD3_bff34518-7181-4f44-8fee-527aa90eb9f4.png]<https://youtu.be/YwtzGvBc4XY>

IMPORTANT NOTICE AND DISCLAIMER Please consider the environment before printing this email. This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the TRT WORLD INFORMATION SECURITY unit at information.security@trtworld.com <mailto:information.security@trtworld.com>. This message contains confidential information and is intended only for the individual or individuals named within. If you are not the named addressee you should not disseminate, distribute or copy this e-mail. Please notify the sender immediately by e-mail if you have received this e-mail by mistake and delete this e-mail from your system. If you are not the intended recipient you are notified that disclosing, copying, distributing or taking any action in reliance on the contents of this information is strictly prohibited. TRT WORLD has taken reasonable precautions to ensure this e-mail and any attachments are free from any viruses and it will remain your responsibility to scan this e-mail and any attachments for viruses. To the extent permitted by law, TRT WORLD do not accept any liability (whether in contract, negligence or otherwise) for any virus infection and/or external compromise of security, breach of confidentiality in relation to transmissions sent by e-mail and/or loss or damage arising from the use of this email or attachments. Any views expressed by an individual within this email do not necessarily reflect the views of TRT WORLD. We do not accept email as a form of notice.

**Comment by paulfitz** [comment]
There's a promise to post pretrained models when they hit a quality target at https://github.com/mozilla/DeepSpeech/issues/435 (I'm not affiliated with this project).

**Comment by elpimous** [comment]
Question : I work on small corpus, for robotic scenarios..
Can I add new material to existing one, and CONTINUE training (with use of existing checkpoint) ? Or must I, due to new material, restart from beginning..
Thanks all

**Comment by mertyildiran** [comment]
@fzliu I think https://github.com/mozilla/DeepSpeech/issues/711 is answering your question(partially) and making this issue redundant. I'm also waiting the release of the official pre-trained models. I think anyone who are looking for the news about pre-trained models should be focused on https://github.com/mozilla/DeepSpeech/issues/711.

**Comment by gvoysey** [comment]
@elpimous you should probably open a new issue for that, but, yes, you can.  You have to fiddle around a bit to use a new export directory but the same checkpoint directory, but i believe that serial training on different corpuses is possible.  Certainly it produced _some_ output graph for me that wasn't totally wrong.

**Comment by fzliu** [comment]
Found this tidbit in the 2017-08-21 meeting notes:

```
Packaging model for "alpha soft release"
Baby step TED+Librivox+Fisher+SWD training [Kelly]
Baby step verify deployment infra works [Alex]
Baby step NPR importer [Kelly]
Baby step TED+Librivox+Fisher+SWD+NPR training [Kelly]
```

So I presume a public pre-trained model is coming soon?

**Comment by est31** [comment]
@fzliu it seems so: https://github.com/mozilla/DeepSpeech/issues/821#issuecomment-328015943

**Comment by fzliu** [comment]
Seems that a publicly available model is coming soon (with LMs!):

```
Packaging for alpha release, what needs to be done? (Unblocking community)
Setup testing of model on TaskCluster ("Should be easy" -Alex) [Alex]
Write script that loads checkpoint + does inference [Reuben]
Write documentation [Kelly]
Decide where to store models (Language Models + DeepSpeech Models) [gitlfs (LM) + release (GitHub Release)]
Native client binaries how they're obtained + installed? [gitlfs, s3...]
Release should contain everything: language models + DeepSpeech model + code...
Decide how to package: models + frozen model
Include a demo? (native client, Reuben’s GUI merged after release [Reuben])
```

Now we just have to wait. Can one or more of the folks at Mozilla let us know when this is done? Many thanks.

**Comment by kdavis-mozilla** [comment]
We'll let you know!

**Comment by WilliamTambellini** [comment]
any ETA ? Thanks.

**Comment by reith** [comment]
I just published [some tools and models](https://github.com/reith/deepspeech-playground) based on Baidu's Deepspeech implementation; It's unidrectional RNNs and without language model I reached %5 WER on LibriSpeech `test-clean`.. Feel free to use pre-trained models, I hope other people share weights too, I have no idea how good are we now.

Sorry, the code is not much clean since I just wanted to try some of my ideas..

**Comment by nealmcb** [comment]
For those looking to stay up-to-date, here's a helpful link I found that includes some of the excerpts quoted above, along with more recent weekly updates:

 [DeepSpeech Weekly Meeting Notes · mozilla/DeepSpeech Wiki](https://github.com/mozilla/DeepSpeech/wiki/Meeting-Notes)

**Comment by reuben** [comment]
We have a first release of an American English model available on https://github.com/mozilla/DeepSpeech/releases/latest

Please check it out and experiment, we're excited to see what you can do with it. We also set up discussion forums on Discourse, check the release notes for links.

**Comment by fotiDim** [comment]
@reuben can we get some details on which exact dataset(s) you used to train the model?

**Comment by reuben** [comment]
@fotiDim Librispeech, Fisher and Switchboard (see our importers for the splits).

**Comment by zh794390558** [comment]
> Librispeech, Fisher and Switchboard (see our importers for the splits).

@fotiDim I think the WER and training datasets should be in wiki too.
